linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -315,6 +315,19 @@ private final class IRChecker(unit: LinkingUnit, reporter: ErrorReporter,
         typecheckExpect(thenp, env, tpe)
         typecheckExpect(elsep, env, tpe)
 
+      case LinkTimeIf(cond, thenp, elsep) if featureSet.supports(FeatureSet.LinkTimeNodes) =>
+        val tpe = tree.tpe
+        // the `cond` is entirely checked in ClassDefChecker
+        coreSpec.linkTimeProperties.tryEvalLinkTimeBooleanExpr(cond) match {
+          case Some(value) =>
+            if (value)
+              typecheckExpect(thenp, env, tpe)
+            else
+              typecheckExpect(elsep, env, tpe)
```

Comment:
It might be worth a comment why we cannot go down both branches here. (IIUC we might not have relevant type information loaded, both branches should be structurally correct).